opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -238,18 +238,22 @@ def set_attribute(self: 'Span',
 
     def add_event(self: 'Span',
                   name: str,
-                  attributes: 'types.Attributes',
+                  attributes: 'types.Attributes' = None,
                   ) -> None:
         if self.events is Span.empty_events:
             self.events = BoundedList(MAX_NUM_EVENTS)
+        if attributes is None:
+            attributes = Span.empty_attributes
```

Comment:
It somewhat irks me that we can't use `Span.empty_attributes` as default and leave out this check, because the API already specifies a default argument of `None`. But I know of no better way to write this than the one you used here.

Reply:
I initially tried to use `{}` as default value but the CI complained with:
```
Dangerous default value {} as argument
```
and I read recommendations to use `None` in that case.